checklists/appsvc_checklist.json

```diff
@@ -0,0 +1,304 @@
+{
+  "$schema": "checklist.schema.json",
+  "items": [
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Outbound network access should be controlled",
+      "description": "Control outbound network access using a combination of regional VNet integration, network security groups and UDR's.  Traffic should be routed to an NVA such as Azure Firewall.  Ensure to monitor the Firewall's logs.",
+      "guid": "c12159e1-14b9-433d-b574-ecccd9bd3baf",
+      "severity": "Medium",
+      "link": "https://docs.microsoft.com/azure/app-service/overview-vnet-integration"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Ensure a stable IP for outbound communications towards internet adresses",
+      "description": "You can provide a stable outbound IP by using VNet integration and using a VNet NAT Gateway or an NVA like Azure Firewall.  This allows the receiving party to allow-list based on IP, should that be needed.  Note that for communications towards Azure Services often there's no need to depend on the IP address and mechanics like Service Endpoints should be used instead.  (Also the use of private endpoints on the receiving end avoids for SNAT to happen and provides a stable outbound IP range.)",
+      "guid": "cda3b54e-b2eb-403d-b9a2-582718d2ddb1",
+      "severity": "Low",
+      "link": "https://docs.microsoft.com/azure/app-service/networking/nat-gateway-integration"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Inbound network access should be controlled",
+      "description": "Control inbound network access using a combination of App Service Access Restrictions, Service Endpoints or Private Endpoints.",
+      "guid": "0725769e-e669-41a4-a34a-c932223ece80",
+      "severity": "High",
+      "link": "https://docs.microsoft.com/azure/app-service/networking-features#access-restrictions"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Use a WAF in front of App Service",
+      "description": "Protect against malicious inbound traffic using a Web Application Firewall like Application Gateway or Azure Front Door.  Make sure to monitor the WAF's logs.",
+      "guid": "b123071a-5416-4415-a33e-a3ad2c2de732",
+      "severity": "High",
+      "link": "https://docs.microsoft.com/azure/app-service/networking/app-gateway-with-service-endpoints"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Avoid for WAF to be bypassed",
+      "description": "Make sure the WAF cannot be bypassed by locking down access to only the WAF.  Use a combination of Access Restrictions, Service Endpoints and Private Endpoints.",
```

Comment:
Should an addition be made here for header and FDID check for AFD?

Reply:
Hmm, it's mentioned in the linked doc and I wouldn't add too much detail in the checklist itself.